hazelcast/src/main/java/com/hazelcast/map/impl/operation/SetOperation.java

```diff
@@ -32,17 +32,19 @@ public SetOperation(String name, Data dataKey, Data value, long ttl) {
 
     @Override
     public void afterRun() {
-        eventType = newRecord ? EntryEventType.ADDED : EntryEventType.UPDATED;
+        eventType = response ? EntryEventType.ADDED : EntryEventType.UPDATED;
         super.afterRun();
     }
 
     @Override
     public void run() {
-        newRecord = recordStore.set(dataKey, dataValue, ttl);
+        Object oldValue = recordStore.put(dataKey, dataValue, ttl);
```

Comment:
`recordStore.set` should be used here because `recordStore.put` loads old-value from map-store.